File: Backend/src/main/java/rc/bootsecurity/requestModule/ticketModule/entity/Ticket.java
package rc.bootsecurity.requestModule.ticketModule.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import rc.bootsecurity.requestModule.commonModule.entity.Request;

import javax.persistence.*;

@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "tbl_tickets")
@Data
@PrimaryKeyJoinColumn(name="request_id")
public class Ticket extends Request {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    /**
     * May contain software specific name, but also user's name or "other" type
     */
    @Column(name = "t_application_name")
    private String ticketSubtypeName;

    @Column(name = "t_request")
    private String problem;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "t_type_id")
    private TicketType ticketType;
}

File: Client/src/app/resources/request/model/Ticket.ts
export interface TicketForm {
  ticketType: string;
  ticketSubtypeName: string;
  requestPriority: string;
  name: string;
  problem: string;
}


export interface TicketSubtype{
  id: number,
  name: string,
  sequence: number,
  active: boolean,

  ticketType: TicketType,
}

export interface TicketType{
  id: number,
  name: string,
  sequence: number,
  active: boolean,
}
